libraries/abstractions/platform/freertos/include/platform/iot_network_udp.h

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) 2019 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file iot_network_udp.h
+ * @brief Declares the network stack functions specified in platform/iot_network.h for
+ * Amazon FreeRTOS UDP Sockets.
```

Comment:
Why are we creating a completely new interface rather than using the existing sockets one?

Reply:
Sorry, the branch I was trying to merge this branch into was outdated. Platform related topic should be another PR. This PR is only focusing on UDP socket implementation for TI.

As for the question, the network interface for UDP to mirror the current abstraction layer (in iot_network_freertos.h). Future application may need this abstraction instead of the existing sockets.